`lz4::lz4` is incorrect

When CMake finds `lz4` through `PREFIX/lib/cmake/lz4/lz4Targets.cmake` (and _not_ with `pkg-config`), the namespace isn't called `lz4`, but `LZ4`. At least here on NixOS (although NixOS doesn't do anything weird to the natural build process of `lz4`, [package file for reference](https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/lz/lz4/package.nix)):

```ShellSession
$ # below prefix path is the output of `nix path-info nixpkgs#lz4.dev`
$ grep add_library /nix/store/jqzzn5m9a2zsrrn424i50dpikqm397w0-lz4-1.10.0-dev/lib/cmake/lz4/lz4Targets.cmake
add_library(LZ4::lz4_shared SHARED IMPORTED)
add_library(LZ4::lz4 INTERFACE IMPORTED)
```

See: https://github.com/desktop-app/cmake_helpers/blob/06a8d5570d90eb0fa9918067fe95e6e193603698/external/lz4/CMakeLists.txt#L13-L13

Relevant build logs:
```ShellSession
CMake Error at cmake/external/lz4/CMakeLists.txt:13 (target_link_libraries):
  The link interface of target "external_lz4" contains:

    lz4::lz4

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.
```

Some info about the system I get the build failure on:
 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.26, NixOS, 25.05 (Warbler), 25.05.20250505.3730d8a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.3`
 - nixpkgs: `/nix/store/9rc9abg9f664bjfhzfp4cb8mrwh7b5y4-source`

--- a/external/lz4/CMakeLists.txt
+++ b/external/lz4/CMakeLists.txt
@@ -10,7 +10,7 @@ add_library(desktop-app::external_lz4 ALIAS external_lz4)
 if (DESKTOP_APP_USE_PACKAGED)
     find_package(lz4 QUIET)
     if (lz4_FOUND)
-        target_link_libraries(external_lz4 INTERFACE lz4::lz4)
+        target_link_libraries(external_lz4 INTERFACE LZ4::lz4)
         return()
     endif()
 

